### PR TITLE
Funannotate: add annotate and compare tools

### PR DESCRIPTION
**Pull request by abretaud**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Adding 2 subcommands of funannotate: `annotate` (to add functional annotation info) and `compare` to compare multiple annotations.

Tests are a bit stupid as making real complete tests would require much bigger test datasets.

Also added the --force option for predict tool

### CODE REVIEW

File: tools/funannotate/funannotate_compare.xml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+        <!-- Need this to change path in the test funannotate_db -->
+        <param type="hidden" name="uglyTestingHack" value="" />
+    </inputs>
+    <outputs>
+        <data name='output' format='html' label="${tool.name} on ${on_string}: report" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="input" value="compare/Genus_species.gbk,compare/Other_species.gbk,compare/Other_beast.gbk,compare/Yet_another.gbk" />
```
Comment (bgruening) [inline tools/funannotate/funannotate_compare.xml:111]: but you are not using the uglyTestingHack here?

File: tools/funannotate/funannotate_annotate.xml [hunk 2]
```diff
@@ -0,0 +1,413 @@
+<tool id="funannotate_annotate" name="Funannotate functional annotation" profile="20.01" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@">
+    <description></description>
```
[the remark this answers is not in the document]
Comment (abretaud) [reply tools/funannotate/funannotate_annotate.xml]: Reduced to `Funannotate functional` to differentiate from other funannotate tools

**Comment by bgruening** [comment]
Now it fails :(

**Comment by abretaud** [comment]
Yep, looking into it! The doc does not agree with the code apparently

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Lovely! Thanks!

**Pull request merged**